lib/vsc/install/shared_setup.py

```diff
@@ -263,12 +263,24 @@ def _fvs(msg=None):
 
 
 def _read(source, read_lines=False):
-    """read a file, either in full or as a list (read_lines=True)"""
-    with open(source) as file_handle:
+    """read a file, either in full or as a list (read_lines=True) and decode UTF-8 files into ASCII"""
+
+    def file_read(file_handle, read_lines):
+        """raw read action on file handle, either in full or as a list (read_lines=True)"""
         if read_lines:
             txt = file_handle.readlines()
         else:
             txt = file_handle.read()
+        return txt
+
+    try:
+        with open(source) as file_handle:
+            txt = file_read(file_handle, read_lines)
+    except UnicodeDecodeError:
+        # file contains unicode characters, try again backslashing them
+        with open(source, encoding='ascii', errors="backslashreplace") as file_handle:
```

Comment:
why not only using this open instead of the try/except? also, is this py2 compat?

Reply:
In Python 2 `open()` does not accept `encoding` and `errors`